src/main/java/com/contrastsecurity/maven/plugin/AbstractContrastMojo.java

```diff
@@ -0,0 +1,196 @@
+package com.contrastsecurity.maven.plugin;
+
+import com.contrastsecurity.sdk.ContrastSDK;
+import java.net.Authenticator;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.settings.Settings;
+
+/**
+ * Abstract mojo for mojos that need to connect to Contrast. Handles authentication, organization
+ * selection (multi-tenancy), and proxy configuration.
+ *
+ * <p>Extensions of this class use the {@link #connectToContrast()} to obtain an instance of the
+ * {@link ContrastSDK} with which they can make requests to Contrast.
+ */
+abstract class AbstractContrastMojo extends AbstractMojo {
+
+  @Parameter(defaultValue = "${settings}", readonly = true)
+  private Settings settings;
+
+  /**
+   * User name for communicating with Contrast. Agent users lack permissions required by this
+   * plugin. <a href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your personal
+   * keys</a>
+   */
+  @Parameter(alias = "username", required = true)
+  private String userName;
+
+  /**
+   * API Key for communicating with Contrast. <a
+   * href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your personal keys</a>
+   */
+  @Parameter(property = "apiKey", required = true)
+  private String apiKey;
+
+  /**
+   * Service Key for communicating with Contrast. <a
+   * href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your personal keys</a>
+   */
+  @Parameter(property = "serviceKey", required = true)
+  private String serviceKey;
+
+  /** Contrast API URL */
+  @Parameter(alias = "apiUrl", defaultValue = "https://app.contrastsecurity.com/Contrast")
+  private String url;
+
+  /**
+   * Unique ID for the Contrast Organization to which the plugin reports results. <a
+   * href="https://docs.contrastsecurity.com/en/personal-keys.html">Find your Organization ID</a>
+   */
+  // TODO[JG] must this be required? If a user is only in one org, we can look it up using the
+  // endpoint /ng/profile/organizations
+  @Parameter(alias = "orgUuid", required = true)
+  private String organizationId;
+
+  /**
+   * When true, will override Maven's proxy settings with Contrast Maven plugin specific proxy
+   * configuration
+   *
+   * @deprecated in a future release, we will remove the proprietary proxy configuration in favor of
```

Comment:
I deprecated the proxy configuration because we should not be configuring our own `java.net.Proxy`. We should use Maven's. We introduced this because a user wanted to use different proxy settings for Contrast. This is also achievable with standard Maven proxy configuration by using named proxies. The plugin should accept an optional `proxyName` which identifies a proxy they've configured in their Maven settings.

Reply:
Might be worth throwing the code `@Deprecated` annotations on here too.

Reply:
agreed